param_passing/proposal.md

```diff
@@ -0,0 +1,194 @@
+
+* `modify`: indicates that the values in the original model are to be modified to the new listed values
+* `replace`: replaces the elements and/or values from the original model to the new provided elements and/or values
+* `add`: adds new elements to the original model
+* `remove`: removes (or disables) the elements from the original model
+
+##### Alternatives considered
+
+Alternatively, instead of using the proposed element identifier, the user could use the attribute `id` for each customizable element in the original file but the intent of using the named path (i.e., the name of the parent elements and itself separated by double colons) was to provide a way for the user to modify the original model without requiring large amounts of updates to the original file.
+
+## Example
+
+Let's look at an example, here is an original model `base_robot`:
+
+```xml
+<model name="base_robot">
+  ...
+  <link name="chassis">
+    <visual name="lidar_visual">
+      ...
+    </visual>
+    <sensor name="lidar">
+      ...
+    </sensor>
+    <sensor name="camera">
+      <pose>0.5 0.02 0 0 0</pose>
+      ...
+    </sensor>
+  </link>
+
+  <link name="top">
+    <visual name="camera_visual">
+      ...
+      <geometry>
+        <box>
+          <size>0.03 0.03 0.05</size>
+        </box>
+      </geometry>
+      <material>
+        <ambient>0.0 1.0 0.0 1</ambient>
+        <diffuse>0.0 1.0 0.0 1</diffuse>
+        <specular>0.5 0.5 0.5 1</specular>
+      </material>
+    </visual>
+    <sensor name="camera">
+      <pose>0.2 1 0.5 0 0 0</pose>
+      <update_rate>20</update_rate>
+      ...
+    </sensor>
+  </link>
+```
+
+Now the user would like to use the original model `base_robot` in a world file but with modifications. The modifications include removing the lidar from the chassis link, adding a plugin to the camera sensor in the chassis link, adding a camera visual to the chassis link, update camera information in the top link, and replace the top link's camera visual geometry.
+
+```xml
+<model name="custom_robot">
```

Comment:
Can add the`<sdf>` tag with the namespace definition so this example is a valid xml?

Reply:
84cc9e0cb3e60071a62ce3d0769286de5d2d5464